File: python/src/kidswithCandies.py
# https://leetcode.com/problems/kids-with-the-greatest-number-of-candies

from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        result = [False] * len(candies)

        greatest = max(candies)

        for i, c in enumerate(candies):
            if c + extraCandies >= greatest:
                result[i] = True

        return result

File: go/kidsWithCandies.go
// https://leetcode.com/problems/kids-with-the-greatest-number-of-candies

package leetcode

import (
	"math"
)

func kidsWithCandies(candies []int, extraCandies int) []bool {
	result := make([]bool, len(candies))

	max := math.MinInt
	for _, c := range candies {
		if c > max {
			max = c
		}
	}

	for i, c := range candies {
		if c+extraCandies >= max {
			result[i] = true
		}
	}

	return result
}
